### DISCUSSION TITLE
Bulk Insert in Microsoft.Data.Sqlite

### DISCUSSION THREAD

**Discussion by bricelam** [bricelam/bricelam.github.io#100] [General]
# Bulk Insert in Microsoft.Data.Sqlite

You’ve got a large set of data to import into your SQLite database, but you can’t find the bulk insert API in Microsoft.Data.Sqlite. That’s because there isn’t one! SQLite doesn’t have any special way to bulk insert data.

https://www.bricelam.net/2017/07/20/sqlite-bulk-insert.html

**Comment by bricelam** [comment]
_From **[Brice Lambson](https://disqus.com/by/bricelam)** on July 21, 2017 18:14_

<p>If anyone is interested in trying version 2.1, you can use the <a href="https://dotnet.myget.org/feed/aspnetcore-dev/package/nuget/Microsoft.Data.Sqlite" rel="nofollow noopener" title="https://dotnet.myget.org/feed/aspnetcore-dev/package/nuget/Microsoft.Data.Sqlite">nightly builds</a>.</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Reply by bricelam** [reply]
_From **[Mohamed Bassuny Mohamed](https://disqus.com/by/bassuny3003)** on May 2, 2019 20:51_

<p>if you please can help me in this <br><a href="https://www.codeproject.com/Questions/4044003/How-do-I-make-bulk-insertion-in-sqlite-database-wi" rel="nofollow noopener" title="https://www.codeproject.com/Questions/4044003/How-do-I-make-bulk-insertion-in-sqlite-database-wi">https://www.codeproject.com...</a></p><p>i wil be thankful</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by bricelam** [comment]
_From **[Boris Djurdjevic](https://disqus.com/by/borisdjurdjevic)** on August 19, 2019 16:23_

<p>For EFCore following library supports Bulk Ops:<br><a href="#" rel="nofollow noopener" title="#">https://github.com/borisdj/EFCore.BulkExtensions</a></p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Reply by bricelam** [reply]
_From **[LK](https://disqus.com/by/disqus_RQZ3uEdHbm)** on May 19, 2020 15:12_

<p>Thats pretty neat. I'm not really an EF fan but the MS team are sure trying hard to win me over.</p>

_Copied from original comment on [Disqus](https://disqus.com/home/forum/bricelam)_

**Comment by msft-takend** [comment]
I have an application that needs to log substantial amounts of data to disk very quickly (1.2MS/s of float). Another wrinkle is that it is ARM64 and only has 2 cores.  SQLite (both raw and EFcore) are too slow. Any ideas on what systems might be up to the job on this one.
